src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+}
+
+void EnGo2_CheckCollision(EnGo2* this, GlobalContext* globalCtx) {
+    Vec3s pos;
+    f32 xzDist;
+
+    pos.x = this->actor.posRot.pos.x;
+    pos.y = this->actor.posRot.pos.y;
+    pos.z = this->actor.posRot.pos.z;
+
+    xzDist = D_80A4816C[this->actor.params & 0x1F].xzDist;
+    pos.x += (s16)(xzDist * Math_Sins(this->actor.shape.rot.y));
+    pos.z += (s16)(xzDist * Math_Coss(this->actor.shape.rot.y));
+
+    pos.y += D_80A4816C[this->actor.params & 0x1F].yDist;
+
+    this->collider.dim.pos = pos;
+
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+}
+
+void EnGo2_SwapInitialFrameAnimFrameCount(EnGo2* this) {
+    f32 initialFrame;
+
+    initialFrame = this->skelAnime.initialFrame;
+    this->skelAnime.initialFrame = this->skelAnime.animFrameCount;
+    this->skelAnime.animFrameCount = initialFrame;
+}
+
+s32 func_80A44AB0(EnGo2* this, GlobalContext* globalCtx) {
+    Player* player;
+    f32 arg2;
+
+    player = PLAYER;
+    if ((this->actor.params & 0x1F) == GORON_DMT_BIGGORON) {
+        return false;
+    } else {
+        if ((this->actionFunc != EnGo2_SlowRolling) && (this->actionFunc != EnGo2_ReverseRolling) &&
+            (this->actionFunc != EnGo2_ContinueRolling)) {
+            return false;
+        } else {
+            if (this->collider.base.acFlags & 2) {
+                Audio_PlaySoundGeneral(NA_SE_SY_CORRECT_CHIME, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                this->actor.flags &= ~0x1000000;
+                this->collider.base.acFlags &= ~0x2;
+                EnGo2_StopRolling(this, globalCtx);
+                return true;
+            }
+            if (player->invincibilityTimer <= 0) {
+                this->collider.base.maskA |= 8;
+            } else {
+                return false;
+            }
+            if (this->collider.base.maskB & 1) {
+                this->collider.base.maskB &= ~1;
+
+                arg2 = this->actionFunc == EnGo2_ContinueRolling ? 1.5f : this->actor.speedXZ * 1.5f;
+
+                globalCtx->damagePlayer(globalCtx, -4);
+                func_8002F71C(globalCtx, &this->actor, arg2, this->actor.yawTowardsLink, 6.0f);
+                Audio_PlayActorSound2(player, NA_SE_PL_BODY_HIT);
+                this->collider.base.maskA &= ~0x8;
+            }
+        }
+    }
+    return false;
+}
+
+s32 EnGo2_UpdateWaypoint(EnGo2* this, GlobalContext* globalCtx) {
+    s32 change;
+
+    if (this->path == NULL) {
+        return 0;
+    } else {
+        change = (u8)(this->path->count - 1);
+        if (this->reverse) {
+            this->waypoint--;
+            if (this->waypoint < 0) {
+                this->waypoint = change - 1;
+            }
+        } else {
+            this->waypoint++;
+            if (this->waypoint >= change) {
+                this->waypoint = 0;
+            }
+        }
+        return 1;
+    }
```

Comment:
You can get rid of the `else` block since this does an early return. (Unless you or others think the else is better/more readable)

Reply:
I'm not entirely sure which one is better/more readable. I've been trying to be consistent with the if/else forms, but in this function I think it would be better to remove the else. But I don't have a preference either way

Reply:
I think an early return works better in this case because it's literally being used to exit the function early if a valid path doesn't exist